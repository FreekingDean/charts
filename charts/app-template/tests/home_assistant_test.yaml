tests:
  - it: validates my home-assistant setup
    set:
      stateful: true
      image:
        repository: ghcr.io/onedr0p/home-assistant
        tag: 2024.9.1
      serviceMonitor:
        scheme: https
        path: /api/prometheus
        bearerTokenSecret:
          name: hass-secret
          key: token
      ports:
        http: 8123
      ingress:
        subdomain: hass
        icon: mdi:home-assistant
      persistence:
        config:
          existingClaim: home-assistant-0
          globalMounts:
            - path: /config
        logs:
          type: emptyDir
          globalMounts:
            - path: /config/logs
        tts:
          type: emptyDir
          globalMounts:
            - path: /config/tts
        tmp:
          type: emptyDir
    asserts:
      - matchSnapshot: {}
