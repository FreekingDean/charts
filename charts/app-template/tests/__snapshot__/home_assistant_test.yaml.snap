validates my home-assistant setup:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        reloader.stakater.com/auto: "true"
      labels:
        app.kubernetes.io/component: RELEASE-NAME
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        helm.sh/chart: app-template-0.0.11
      name: RELEASE-NAME
    spec:
      replicas: 1
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app.kubernetes.io/component: RELEASE-NAME
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: RELEASE-NAME
      strategy:
        type: Recreate
      template:
        metadata:
          labels:
            app.kubernetes.io/component: RELEASE-NAME
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: RELEASE-NAME
        spec:
          automountServiceAccountToken: true
          containers:
            - image: ghcr.io/onedr0p/home-assistant:2024.9.1
              livenessProbe:
                failureThreshold: 3
                initialDelaySeconds: 0
                periodSeconds: 10
                tcpSocket:
                  port: 8123
                timeoutSeconds: 1
              name: RELEASE-NAME
              readinessProbe:
                failureThreshold: 3
                initialDelaySeconds: 0
                periodSeconds: 10
                tcpSocket:
                  port: 8123
                timeoutSeconds: 1
              startupProbe:
                failureThreshold: 30
                initialDelaySeconds: 0
                periodSeconds: 5
                tcpSocket:
                  port: 8123
                timeoutSeconds: 1
              volumeMounts:
                - mountPath: /config
                  name: config
                - mountPath: /config/logs
                  name: logs
                - mountPath: /tmp
                  name: tmp
                - mountPath: /config/tts
                  name: tts
          dnsPolicy: ClusterFirst
          enableServiceLinks: false
          hostIPC: false
          hostNetwork: false
          hostPID: false
          serviceAccountName: default
          volumes:
            - name: config
              persistentVolumeClaim:
                claimName: home-assistant-0
            - emptyDir: {}
              name: logs
            - emptyDir: {}
              name: tmp
            - emptyDir: {}
              name: tts
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        app.kubernetes.io/service: RELEASE-NAME
        helm.sh/chart: app-template-0.0.11
      name: RELEASE-NAME
    spec:
      ports:
        - name: http
          port: 8123
          protocol: TCP
          targetPort: 8123
      selector:
        app.kubernetes.io/component: RELEASE-NAME
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: RELEASE-NAME
      type: ClusterIP
  3: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        hajimari.io/icon: mdi:home-assistant
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        helm.sh/chart: app-template-0.0.11
      name: RELEASE-NAME
    spec:
      ingressClassName: traefik
      rules:
        - host: hass.deangalvin.dev
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      number: 8123
                path: /
                pathType: Prefix
      tls:
        - hosts:
            - hass.deangalvin.dev
          secretName: deangalvin-dev-production-tls
  4: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        helm.sh/chart: app-template-0.0.11
      name: RELEASE-NAME
    spec:
      endpoints:
        - bearerTokenSecret:
            key: token
            name: hass-secret
          interval: 1m
          path: /api/prometheus
          port: http
          scheme: https
          scrapeTimeout: 10s
      jobLabel: RELEASE-NAME
      namespaceSelector:
        matchNames:
          - NAMESPACE
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: RELEASE-NAME
          app.kubernetes.io/service: RELEASE-NAME
